interfaces/cython/cantera/thermo.pyx

```diff
@@ -66,22 +66,23 @@ cdef class Species:
             section='species')
 
     """
-    def __cinit__(self, *args, init=True, **kwargs):
+    def __cinit__(self, name=None, composition=None, *args, init=True, **kwargs):
         if init:
-            self._species.reset(new CxxSpecies())
+            if name is not None:
+                name = stringify(name)
+            else:
+                name = stringify("")
+            if composition is not None:
+                composition = comp_map(composition)
+            else:
+                composition = comp_map("")
+            self._species.reset(new CxxSpecies(name, composition))
```

Comment:
If you switch to a lazy calculation of the molecular weight, I think this can all be left as it was.